devito/profiling.py

```diff
@@ -126,48 +92,40 @@ def new(self):
 
     def summary(self, arguments, dtype):
         """
-        Return a :class:`PerformanceSummary` of the tracked sections.
+        Return a :class:`PerformanceSummary` of the profiled sections.
 
         :param arguments: A mapper from argument names to run-time values from which
                           the Profiler infers iteration space and execution times
                           of a run.
         :param dtype: The data type of the objects in the profiled sections. Used
                       to compute the operational intensity.
         """
-
         summary = PerformanceSummary()
-        for itspace, profile in self._sections.items():
-            dims = {i: i.dim.parent if i.dim.is_Stepping else i.dim for i in itspace}
+        for section, data in self._sections.items():
+            # Time to run the section
+            time = max(getattr(arguments[self.name]._obj, section.name), 10e-7)
 
-            # Time
-            time = max(getattr(arguments[self.name]._obj, profile.name), 10e-7)
+            # Number of FLOPs performed
+            ops = data.ops.subs(arguments)
 
-            # Flops
-            itershape = [i.extent().subs(arguments) for i in itspace]
-            iterspace = reduce(operator.mul, itershape)
-            flops = float(profile.ops*iterspace)
-            gflops = flops/10**9
-            gpoints = iterspace/10**9
+            # Number of grid points computed
+            points = data.points.subs(arguments)
 
             # Compulsory traffic
-            datashape = [
-                (dims[i].symbolic_end - dims[i].symbolic_start + 1).subs(arguments)
-                for i in itspace]
-            dataspace = reduce(operator.mul, datashape)
-            traffic = float(profile.memory*dataspace*dtype().itemsize)
+            traffic = float(data.traffic.subs(arguments)*dtype().itemsize)
+
+            # Runtime itershapes
+            itershapes = [tuple(i.subs(arguments) for i in j) for j in data.itershapes]
 
             # Derived metrics
-            oi = flops/traffic
+            gflops = float(ops)/10**9
             gflopss = gflops/time
```

Comment:
Since we're here, I should comment that I sometimes see crashes with these `time`s equal to zero.
Could we do `gflops/time if time > 0 else 0` or similar?

Reply:
yes!

Reply:
hang on, that's weird because of this line above:
```
max(getattr(arguments[self.name]._obj, section.name), 10e-7)
```
so, ``time`` should never be 0 😮

Reply:
In...deed.  I can't remember exactly what it was.  Maybe it was only `traffic`.